### PR TITLE
add cmake support

### PR DESCRIPTION
**Pull request by endingly**
Add cmake support to prepare for adding vcpkg

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        -Wall
+        -Wstrict-prototypes
+        -Wshadow
+        -Wpointer-arith
+        -Wcast-qual
+        -Wcast-align
+        -Wwrite-strings
+        -Wnested-externs
+        -Wredundant-decls
+        -Werror
```
Comment (dg0yt) [inline CMakeLists.txt:39]: Do not generally add `-Werror`. It is fine for developers, CI and controlled platforms. It is a barrier for new platforms, new compiler versions and for users of build recipes (such as vcpkg ports).

File: CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+install(
+    FILES "${PROJECT_BINARY_DIR}/${PROJECT_NAME}Config.cmake"
+    "${PROJECT_BINARY_DIR}/${PROJECT_NAME}ConfigVersion.cmake"
+    DESTINATION share/${PROJECT_NAME}
+)
+
+install(
+    EXPORT "${PROJECT_NAME}Targets"
+    DESTINATION share/${PROJECT_NAME}
+)
```
Comment (dg0yt) [inline CMakeLists.txt:93]: Consider adding a namespace during the export. Namespaced targets can be detected as targets, allowing improve error messages in downstream CMake usage. Simple target names will be passed to the linker, so a missing `find_package` will appear as linker error without good diagnostics.

**Comment by lammertb** [comment]
Thanks for the pull request.

In the past I have been skeptical adding CMake support because it ads an extra layer of complexity and dependencies, but in case of adding to vcpkg I can definitely see the added value.

**Comment by endingly** [comment]
In fact, vcpkg supports adding make projects, but the support is not as strong as cmake.

**Pull request merged**